### PR TITLE
Userlist search

### PR DESCRIPTION
**Pull request by Pawwol1**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ function App() {
     <>
       <Header/>
       <Userlist/>
-      <UserlistSearch/>
+      {/* <UserlistSearch/> */}
```
Comment (paulinaTargosz) [inline src/App.tsx]: add a comment why would you like to keep it ;)

File: src/components/userlistSearch/userlistSearch.tsx [hunk 2]
```diff
@@ -1,9 +1,66 @@
...
+    const [searchValue, setSearchValue] = useState<string>('');
+
+    useEffect(() => {
+        setUserlist(users);
+    },[users, searchValue])
+
+    const handleOnClick = () => {
+        const findUsers = 
+        userlist && userlist?.length > 0 
+            ? userlist?.filter((user) => user?.first_name.toLowerCase().includes(searchValue.toLowerCase()) || user?.last_name.toLowerCase().includes(searchValue.toLowerCase()) || user?.email.toLowerCase().includes(searchValue.toLowerCase()))
```
Comment (paulinaTargosz) [inline src/components/userlistSearch/userlistSearch.tsx]: this line is too long

Comment (paulinaTargosz) [reply src/components/userlistSearch/userlistSearch.tsx]: can you do it more generic?

File: src/App.tsx (hunk 1 above)
Comment (Pawwol1) [reply src/App.tsx]: I don't want to keep it. I forgot to delete it ;)

File: src/components/userlist/userlist.tsx [hunk 3]
```diff
@@ -15,68 +15,36 @@ function Userlist() {
...
+        const res = await fetch(URL);
         if (!res.ok) {
             const err = "User not Found";
             throw new Error(err);
         }
         const data = await res.json();
         setUsers(data.data);
-        // setTotalPages(Array(data.total_pages).fill(1));
         setTotalPages(data.total_pages);
+        console.log(data.data[0])
```
Comment (paulinaTargosz) [inline src/components/userlist/userlist.tsx]: remove

File: src/components/userlistSearch/userlistSearch.tsx [hunk 4]
```diff
@@ -1,9 +1,71 @@
...
+    },[users, searchValue])
+
+    const handleOnClick = () => {
+        const searchToLowerCase = (value: string) => {
+            return value.toLowerCase().includes(searchValue.toLowerCase())
+        }
+
+        const findUsers = 
+        userlist && userlist?.length > 0 
+            ? userlist?.filter((user) => searchToLowerCase(user?.first_name) || searchToLowerCase(user?.last_name) || searchToLowerCase(user?.email))
```
Comment (paulinaTargosz) [inline src/components/userlistSearch/userlistSearch.tsx:24]: much better! <3

**Review by paulinaTargosz — APPROVED** [review]

**Pull request merged**